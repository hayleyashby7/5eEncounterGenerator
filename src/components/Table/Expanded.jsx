/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import Monster from '../Monster/Monster';

function Expanded({ data, colSpan }) {
    return (
        <tr>
            <td colSpan={colSpan}>
                <Monster data={data} />
            </td>
        </tr>
    );
}

Expanded.propTypes = {
    data: PropTypes.object.isRequired,
    colSpan: PropTypes.number.isRequired,
};

export default Expanded;
