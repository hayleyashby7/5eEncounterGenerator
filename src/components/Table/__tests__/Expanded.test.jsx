import { it, describe, expect, vi, beforeEach } from 'vitest';
import React from 'react';
import { render } from '@testing-library/react';
import Expanded from '../Expanded';

vi.mock('../../Monster/Monster');

describe('Expanded', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    it('should render the component', async () => {
        // Arrange
        const mockMonster = await import('../../Monster/Monster');
        mockMonster.default.mockImplementation(() => <div>Monster</div>);
        const tbody = document.createElement('tbody');

        // Act
        const { container } = render(<Expanded data={{}} colSpan={1} />, {
            container: document.body.appendChild(tbody),
            baseElement: document.body,
        });

        // Assert
        expect(container).toMatchSnapshot();
        expect(mockMonster.default).toHaveBeenCalledTimes(1);
    });
});
