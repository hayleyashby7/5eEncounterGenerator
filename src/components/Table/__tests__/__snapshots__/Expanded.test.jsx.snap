// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Expanded > should render the component 1`] = `
<tbody>
  <tr>
    <td
      colspan="1"
    >
      <div>
        Monster
      </div>
    </td>
  </tr>
</tbody>
`;
